### ISSUE TITLE
Add documentation

### ISSUE THREAD

**Issue by antoyo** [antoyo/tql#1]

**Comment by Ygg01** [comment]
Hm. I'll try to get started on this as soon as possible.

I'm leaving some resources for further digestion: 
* [Steve Losh - Anatomy of Good documentation](http://stevelosh.com/blog/2013/09/teach-dont-tell/#the-anatomy-of-good-documentation)
* [Daniele Procida - What nobody tells you about documentation](https://www.divio.com/en/blog/documentation/)

**Comment by antoyo** [comment]
Thank you.
If you need any help, don't hesitate to ask questions.
I took a break of coding and it will continue for the next weeks (I need to rest my wrists), but I can answer your questions :) .

**Comment by Ygg01** [comment]
I can use whatever (although I'll need to install Ruby etc). One problem with code written in outside tools, is that we can't test it in Rust.

**Comment by antoyo** [comment]
That's a very good point, I forgot about that.
Is there some tools to build a website that can test the Rust code?

**Comment by Ygg01** [comment]
I don't think so. There is [`mdBook`](https://rust-lang-nursery.github.io/mdBook/cli/test.html). But that's written in Markdown. Maybe we could write AsciiDoc parser for `mdBook` :P

**Comment by antoyo** [comment]
I'll look at that but I'm not sure I like the layout of the resulting HTML.
Maybe we can configure it.

**Comment by antoyo** [comment]
I'll write a tool to check the code in asciidoctor files, probably using rustdoc.
So, please use asciidoctor to write the documentation.
For code comments, use the following syntax for now:
```adoc
[source,rust]
// # extern crate tql;
fn main() {
}
```
(That is, without the delimiters `----` to be able to insert comments. I asked [here](https://github.com/asciidoctor/asciidoctor/issues/2516) if there's another way to do so.)
where the line `// #` is like lines starting with `#` in the `rustdoc` examples syntax.

If you need any help with asciidoctor, don't hesitate to ask.
Also, I'll setup a basic website in the [tql-website repository](https://github.com/antoyo/tql-website) that will be automatically updated when asciidoctor files are updated.

**Comment by antoyo** [comment]
I started to setup the website in [this repository](https://github.com/antoyo/tql-website), so that you can create pull requests in this repository with the documentation.
While there's still content in Markdown, I'll convert everything to Asciidoctor when I get back from vacation, so please use asciidoctor as well.
[Here's the current URL of the website.](https://antoyo.github.io/tql-website/) I'll register a domain for this website when it's ready to be shown to the public.
I still have some more work to do on this website.
Thanks.

**Comment by Ygg01** [comment]
@antoyo Most excellent. I apologise for late response, but I was in process of migrating some data from old -> new PC. I'll get on it ASAP, but I'm going home for holidays, so I'm unsure how much I'll be able to contribute before Monday. I'll fork it and try some stuff out.

**Comment by antoyo** [comment]
No problem.
The syntax is going to change (temporarily, hopefully) when PR #23 is merged.
